x-pack/plugins/infra/public/hooks/use_breadcrumbs.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ChromeBreadcrumb } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { MouseEvent, useEffect } from 'react';
+import { EuiBreadcrumb } from '@elastic/eui';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+function handleBreadcrumbClick(
+  breadcrumbs: ChromeBreadcrumb[],
+  navigateToHref?: (url: string) => Promise<void>
+) {
+  return breadcrumbs.map((bc) => ({
+    ...bc,
+    ...(bc.href
+      ? {
+          onClick: (event: MouseEvent) => {
+            if (navigateToHref && bc.href) {
+              event.preventDefault();
+              navigateToHref(bc.href);
+            }
+          },
+        }
+      : {}),
+  }));
+}
+
+type AppId = 'logs' | 'metrics';
+
+export const makeBaseBreadcrumb = (
+  path: string,
+  observabilityPath: string,
+  app: AppId
+): [EuiBreadcrumb, EuiBreadcrumb] => {
+  const appText =
+    app === 'logs'
```

Comment:
I think it would be beneficial to split these out into two separate hooks, one for `useLogsBreadcrumbs` and one for `useMetricsBreadcrumbs`, those hooks could then wrap this hook and pass in the app name / app href as a parameter, rather than having the ternary internally. We're leaning more and more towards splitting logs and metrics now (even if it seems a little laborious in the code without the official split).

Reply:
Good suggestion, I split into two separate hooks as suggested!